packages/apps/explorer/src/components/search/search.tsx

```diff
@@ -0,0 +1,237 @@
+import { truncateValues } from '@/services/format';
+import { MonoSearch } from '@kadena/react-icons/system';
+import { Badge, Box } from '@kadena/react-ui';
+import { atoms } from '@kadena/react-ui/styles';
+import React, { useState } from 'react';
+
+export type SearchItemTitle =
+  | 'Account'
+  | 'Request Key'
+  | 'Block Height'
+  | 'Block Hash'
+  | 'Events';
+
+export interface ISearchItem {
+  title: SearchItemTitle;
+  disabled?: boolean;
+}
+interface ISearchComponentProps {
+  placeholder: string;
+  searchItems: ISearchItem[];
+}
+
+const SearchCombobox: React.FC<ISearchComponentProps> = ({
+  placeholder,
+  searchItems,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [searchOption, setSearchOption] = useState<number | null>(null);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [optionClicked, setOptionClicked] = useState(false);
+  const [escapePressed, setEscapePressed] = useState(false);
+
+  const setOptionsDisabledExcept = (exceptIndex: number): void => {
+    searchItems.forEach((item, index) => {
+      if (index !== exceptIndex) {
+        item.disabled = true;
+      }
+    });
+  };
+
+  const inferOption = (value: string): SearchItemTitle | undefined => {
+    if (
+      value.toLocaleLowerCase().startsWith('k:') ||
+      value.toLocaleLowerCase().startsWith('w:')
+    ) {
+      return 'Account';
+    } else if (value.includes('.')) {
+      return 'Events';
+    } else if (value.length === 43) {
+      return 'Request Key';
+    } else if (/^\d+$/.test(value)) {
+      return 'Block Height';
+    }
+
+    return undefined;
+  };
+
+  const enableAllOptions = (): void => {
+    searchItems.forEach((item) => {
+      item.disabled = false;
+    });
+  };
+
+  const handleSearch = (value: string, option: number | null): void => {};
+
+  const handleSearchValueChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
+    setSearchValue(e.target.value);
+
+    if (escapePressed || optionClicked) return;
+
+    const inferedOption = inferOption(e.target.value);
+    if (inferedOption === 'Account') {
+      setSearchOption(0);
+      setOptionsDisabledExcept(0);
+    }
+    if (inferedOption === 'Request Key') {
+      setSearchOption(1);
+      setOptionsDisabledExcept(1);
+    }
+
+    if (inferedOption === 'Block Height') {
+      setSearchOption(2);
+      setOptionsDisabledExcept(2);
+    }
+
+    if (!inferedOption || inferedOption === undefined) {
+      setSearchOption(null);
+      enableAllOptions();
+    }
+  };
+
+  const handleSearchValueKeyDown = (
```

Comment:
Can you double check if this also works on mobile? I feel that the key names might differ there.